src/components/compress/index.tsx

```diff
@@ -100,12 +100,22 @@ async function preprocessImage(
 ): Promise<ImageData> {
   let result = source.processed;
 
+  if (isWorkerResizeOptions(preprocessData.resize) && isHqx(preprocessData.resize)) {
+    result = await processor.workerResize(result, preprocessData.resize);
```

Comment:
Could this be moved into the `else if (isHqx(preprocessData.resize)) {` block below?